pkg/cmd/sourcemapdiff/options_handler.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/grafana/plugin-validator/pkg/archivetool"
+	"github.com/grafana/plugin-validator/pkg/repotool"
+)
+
+func getLocalPathFromSourceCodeOption(sourceCodeUri string) (string, func(), error) {
+	sourceCodePath, sourceCodeCleanup, err := sourceCodeUriToLocalPath(sourceCodeUri)
+	if err != nil {
+		return "", nil, err
+	}
+
+	// check if there's a src/plugin.json file
+	pluginJSONPath := filepath.Join(sourceCodePath, "src", "plugin.json")
+	if _, err := os.Stat(pluginJSONPath); os.IsNotExist(err) {
+		return "", nil, fmt.Errorf("src/plugin.json file not found in %s", pluginJSONPath)
+	}
```

Comment:
I'll remove the IsNotExist check and leave only the error check that prints the stat error. That one will already say that the error is because the file does not exist so I don't repeat logic. Considering the end user for this is a technical one this should be enough to hint them about the problem.